src/convenience.jl

```diff
@@ -0,0 +1,186 @@
+"""
+    geoarray(filename; kw...) => AbstractGeoArray
+
+Load a file path as `AbstractGeoArray`. 
+
+# Keywords
+
+Passed to the constructor for the file type, and commmonly include:
+
+- `crs`: crs to use instead of the detected crs.
+- `mappedcrs`: CRS format like `EPSG(4326)` used in `Selectors` like `Between` and `At`, and
+    for plotting. Can be any CRS `GeoFormat` from GeoFormatTypes.jl, like `WellKnownText`.
+- `name`: `Symbol` name for the array.
+- `dims`: `Tuple` of `Dimension`s for the array. Detected automatically, but can be passed in.
+- `refdims`: `Tuple of` position `Dimension`s the array was sliced from.
+- `missingval`: Value reprsenting missing values. Detected automatically when possible, but
+    can be passed it.
+- `metadata`: `Metadata` object for the array. Detected automatically but can be passed in.
+"""
+function geoarray end
+
+geoarray(filename::AbstractString; kw...) = _constructor(geoarray, filename)(filename; kw...)
+
+"""
+    Base.write(filename::AbstractString, A::AbstractGeoArray; kw...)
+
+Write any [`AbstractGeoArray`](@ref) to file, guessing the backend from the file extension.
+
+Keyword arguments are passed to the `write` method for the backend.
+"""
+function Base.write(
+    filename::AbstractString, A::AbstractGeoArray; kw...
+)
+    write(filename, _constructor(geoarray, filename), A; kw...)
+end
+
+"""
+    stack(filename; kw...) => AbstractGeoStack
+    stack(filenames::NamedTuple; kw...) => AbstractGeoStack
+
+Load a file path or a `NamedTuple` of paths as an `AbstractGeoStack`.
+
+# Arguments
+
+- `filename`: A `NamedTuple` of stack keys and `String` filenames.
+
+# Keywords
+
+Passed to the constructor for the file type, and commmonly include:
+
+- `window`: A `Tuple` of `Dimension`/`Selector`/indices that will be applied to the
```

Comment:
Should/could `geoarray` have this keyword too?

Reply:
It's been a question... for geoarray it kind of doubles up `view`, so it could be better if `view` actually worked on `DiskGeoArray` lazily - but we need another wrapper layer to achieve that. `window` is like a `view` for an uninstantiated object - because `view` has to wrap an `AbstractArray` and we don't have that yet in a stack.

Reply:
Thanks, makes sense.